kernel/libs/aster-util/src/segment_slice.rs

```diff
@@ -2,41 +2,41 @@
 
 // SPDX-License-Identifier: MPL-2.0
 
-//! Provides [`SegmentSlice`] for quick duplication and slicing over [`Segment`].
+//! Provides [`SegmentSlice`] for quick duplication and slicing over [`Segment<dyn UntypedMeta>`].
 
 use alloc::sync::Arc;
 use core::ops::Range;
 
 use ostd::{
     mm::{
-        FallibleVmRead, FallibleVmWrite, Frame, Infallible, Paddr, Segment, VmIo, VmReader,
-        VmWriter, PAGE_SIZE,
+        FallibleVmRead, FallibleVmWrite, Frame, Infallible, Paddr, Segment, UntypedMem,
+        UntypedMeta, VmIo, VmReader, VmWriter, PAGE_SIZE,
     },
     Error, Result,
 };
 
-/// A reference to a slice of a [`Segment`].
+/// A reference to a slice of a [`Segment<dyn UntypedMeta>`].
 ///
 /// Cloning a [`SegmentSlice`] is cheap, as it only increments one reference
-/// count. While cloning a [`Segment`] will increment the reference count of
+/// count. While cloning a [`Segment<dyn UntypedMeta>`] will increment the reference count of
 /// many underlying pages.
 ///
 /// The downside is that the [`SegmentSlice`] requires heap allocation. Also,
-/// if any [`SegmentSlice`] of the original [`Segment`] is alive, all pages in
-/// the original [`Segment`], including the pages that are not referenced, will
+/// if any [`SegmentSlice`] of the original [`Segment<dyn UntypedMeta>`] is alive, all pages in
+/// the original [`Segment<dyn UntypedMeta>`], including the pages that are not referenced, will
 /// not be freed.
 #[derive(Debug, Clone)]
 pub struct SegmentSlice {
-    inner: Arc<Segment>,
+    inner: Arc<Segment<dyn UntypedMeta>>,
```

Comment:
I have been seeing `Frame<dyn UntypedMeta>` and `Segment<dyn UntypedMeta>` all over the place. These names are too long and too cumbersome for such a commonly used primitive. This makes me question the wisdom of abandoning `Pages<_>` and `ContPages<_>`. How about we keep `Pages<_>` and `ContPages<_>`, but provide `Frame` and `Segment` as type alises?

```rust
type Frame<M = dyn UntypedMeta> = Page<M: UntypedMeta>;
type Segment<M = dyn UntypedMeta> = ContPages<M: UntypedMeta>;
```

This way, user code like `SegmentSlice` can be left unchanged.

```rust
pub struct SegmentSlice {
    inner: Arc<Segment>,
}
```

If you really think about it, untyped memory pages are so widely used that they do deserve special names like frames and segments so that people do not have to say and type untyped frames and untyped segments every time they refer to untyped memory. That would be silly.

Reply:
My concern is, that if the two terms "page" and "frame" are brought together, are generally accepted as virtual pages and physical pages. We are bringing some confusions in your suggested way (`Frame` vs `Page`, `Segment` vs `ContPages`).

Since we are establishing new practices, I suggest some of the following namings:

 - `Frame<M>`/`type AnyFrame = Frame<dyn FrameMeta>` (generic frames) vs.  `UFrame<UM>`/`type AnyUFrame = UFrame<dyn UntypedFrameMeta>` (untyped frames);
 - `Segment<M>`/`type AnySegment = Segment<dyn FrameMeta>` (generic frames) vs.  `USegment<UM>`/`type AnyUSegment = USegment<dyn UntypedFrameMeta>` (untyped segments);